src/jit/emitxarch.cpp

```diff
@@ -13663,6 +13663,1128 @@ size_t emitter::emitOutputInstr(insGroup* ig, instrDesc* id, BYTE** dp)
+            result.insThroughput = PERFSCORE_THROUGHPUT_3C;
+            result.insLatency    = PERFSCORE_LATENCY_11C;
+            break;
+
+        case INS_divsd:
+            result.insThroughput = PERFSCORE_THROUGHPUT_4C;
+            result.insLatency    = PERFSCORE_LATENCY_13C;
+            break;
+
+        case INS_sqrtss:
+            result.insThroughput = PERFSCORE_THROUGHPUT_3C;
+            result.insLatency    = PERFSCORE_LATENCY_12C;
+            break;
+
+        case INS_sqrtsd:
+            result.insThroughput = PERFSCORE_THROUGHPUT_6C;
+            result.insLatency    = PERFSCORE_LATENCY_16C;
+            break;
+
+        case INS_roundsd:
+            result.insThroughput = PERFSCORE_THROUGHPUT_1C;
+            result.insLatency    = PERFSCORE_LATENCY_8C;
+            break;
+
+        case INS_cvttsd2si:
+        case INS_cvtsd2si:
+        case INS_cvttss2si:
+        case INS_cvtss2si:
+            result.insThroughput = PERFSCORE_THROUGHPUT_1C;
+            result.insLatency    = PERFSCORE_LATENCY_6C;
+            break;
+
+        case INS_cvtsd2ss:
+        case INS_cvtps2pd:
+            result.insThroughput = PERFSCORE_THROUGHPUT_1C;
+            result.insLatency    = PERFSCORE_LATENCY_5C;
+            break;
+
+        case INS_cvtsi2sd:
+        case INS_cvtsi2ss:
+            result.insThroughput = PERFSCORE_THROUGHPUT_2C;
+            result.insLatency    = PERFSCORE_LATENCY_6C;
+            break;
+
+        case INS_cvtss2sd:
+            result.insThroughput = PERFSCORE_THROUGHPUT_2C;
+            result.insLatency    = PERFSCORE_LATENCY_5C;
+            break;
+
+        case INS_movaps:
+        case INS_movups:
+        case INS_movapd:
+        case INS_movupd:
+            if (memAccessKind == PERFSCORE_MEMORY_NONE)
+            {
+                result.insThroughput = PERFSCORE_THROUGHPUT_4X;
+                result.insLatency    = PERFSCORE_LATENCY_1C;
+            }
+            else if (memAccessKind == PERFSCORE_MEMORY_READ)
+            {
+                result.insThroughput = PERFSCORE_THROUGHPUT_2X;
+                result.insLatency    = max(PERFSCORE_LATENCY_2C, result.insLatency);
+            }
+            else
+            {
+                assert(memAccessKind == PERFSCORE_MEMORY_WRITE);
+                result.insThroughput = PERFSCORE_THROUGHPUT_1C;
+                result.insLatency    = max(PERFSCORE_LATENCY_3C, result.insLatency);
+            }
+            break;
+
+        case INS_andps:
+            if (memAccessKind == PERFSCORE_MEMORY_NONE)
+            {
+                result.insThroughput = PERFSCORE_THROUGHPUT_3X;
+                result.insLatency    = PERFSCORE_LATENCY_1C;
+            }
+            else
+            {
+                result.insThroughput = PERFSCORE_THROUGHPUT_2X;
+            }
+            break;
+
+        case INS_lzcnt:
+        case INS_tzcnt:
+        case INS_popcnt:
+        case INS_crc32:
+        case INS_pdep:
```

Comment:
`pdep` and `pext`, in particular, are significantly worse perf on AMD Ryzen machines (at 18 latency/throughput).

Reply:
Is their an alternative sequence of instructions that executes faster that is used on AMD Ryzen?
If not it then we might as well just model the Intel Skylake costs.

Reply:
I don't think so, and I also don't think this is used outside HWIntrinsics today. I just wanted to ensure the few instructions that are significantly different were called out (the vast majority are similar latency/throughput between Skylake and Ryzen, according to Agner's).